### ISSUE TITLE
🐛 Training Opportunities Description doesn't have header styling like in figma 

### ISSUE THREAD

**Issue by brindasasi** [GCTC-NTGC/gc-digital-talent#12086]
## 🐛 Bug

The description text area for Training Opportunities doesn't have option to provide a header in the text. 
So it isn't accessible to the screen reader user and visually look less appealing with lot of text 

![image](https://github.com/user-attachments/assets/507cc06d-03eb-49d4-aac3-d2f7df8ddd7b)

## 🦋 Expected Behaviour

![image](https://github.com/user-attachments/assets/34ed297b-e7c6-4563-8aa2-1c9ec1e6120e)

## 🕵️ Details

We need to add header format icon on the Text area for Training Opportunities Description.
Upon clicking on it , the selected text should be formatted to a header. To simplify we could add only one type of header probably h3? 

Other elements in this page are wrapped by <FieldDisplay> component. Could we use the same while clicking on an icon? 

## ✅ Acceptance Criteria

- [ ] Allow `h3` headers to be added in rich text editor, but ONLY on the training opportunities form
- [ ] These headers are displayed correctly where training opportunities are displayed

**Comment by tristan-orourke** [comment]
I don't think we want to give end users the ability to add `<h>` elements to a page. There's no guarantee the results would follow a proper heading hierarchy. Instead, can we simply add Bold text formatting?

**Comment by brindasasi** [comment]
Bold would be a problem while reading with screen reader I guess.. It wouldn't distinguish . I'm unsure if thats acceptable
 @esizer ?

**Comment by esizer** [comment]
> Bold would be a problem while reading with screen reader I guess.. It wouldn't distinguish . I'm unsure if thats acceptable
> @esizer ?

I sadly agree with both of you :sweat_smile: This is a perpetual issue with giving users tools for formatting since that formatting has semantic meaning.

Unless we have some sort of checker in place (difficult to implement) or do extensive training and reviewing of content before it is published, we will always have issues with users misusing the formatting options we give them and was a key reason why we are strict about what options we give them.

Unfortunately, I do not have a good solution. The way I have handled this in the past may be untenable for us. I would do a 2-3 hour training session on accessibility for web content with content editors before they were given access and then follow up with a full review of new content every month to remediate any issues and found that was one of the only ways to guarantee the site was as close to guidelines as possible. Of course, in between reviews, the site would usually have errors.
